src/components/hooks.js

```diff
@@ -231,29 +199,51 @@ export function useClosestVitessceContainerSize(ref) {
   return [width, height];
 }
 
-export function useExpressionValueGetter({ attrs, expressionData }) {
+export function useExpressionValueGetter({ instanceObsIndex, matrixObsIndex, expressionData }) {
   // Get a mapping from cell ID to row index in the gene expression matrix.
-  const cellIdMap = useMemo(() => {
-    const result = {};
-    if (attrs && attrs.rows) {
-      // eslint-disable-next-line no-plusplus
-      for (let i = 0; i < attrs.rows.length; i++) {
-        result[attrs.rows[i]] = i;
-      }
+  // Since the two obsIndices (instanceObsIndex = the obsIndex from obsEmbedding)
```

Comment:
The edge case here being if you have two different `AnnData` stores or something?

Reply:
Yes, or maybe an AnnData store holds the matrix but a CSV or JSON file holds the embedding coordinates, or vice versa.